docs/tutorials/articles/sales_dashboard/step_03/step_03.md

```diff
@@ -0,0 +1,73 @@
+---
+hide:
+  - toc
+---
+
+The full code for this step is available 
+[here](https://github.com/AlexandreSajus/taipy-course/tree/main/4_charts)
+
+In this part we will embed a Plotly map figure in our application.
+
+![Map embedded in application](images/map.png){ width=90% : .tp-image-border }
+
+For this purpose, we will use the `generate_map` function defined 
+[here](https://github.com/AlexandreSajus/taipy-course/blob/main/4_charts/chart.py)
+to return a Plotly map figure.
+
+![Plotly Map](images/plotly_map.png){ width=50% : .tp-image-border }
+
+Using the same code as the previous steps, we can import the function and initialize the chart:
+
+```python
+from chart import generate_map
+
+data = pd.read_csv("data.csv")
+map_fig = generate_map(data)
+```
+
+We can now add the map to the page by replacing our previous chart and table with:
+
+```python
+        tgb.html("br")
+        with tgb.layout(columns="2 3"):
+            tgb.chart(
+                data="{chart_data}",
+                x="State",
+                y="Sales",
+                type="bar",
+                layout="{layout}",
+            )
+            tgb.chart(figure="{map_fig}")
+        tgb.html("br")
+        tgb.table(data="{data}")
+```
+
+We should now update the callback function to refresh the map when filters are applied:
+
+```python
+def apply_changes(state):
+    state.data = data[
```

Comment:
Keep data until all the changes are done to data. Then, state.data = data

Reply:
Agreed